app/src/test/java/q4/team1/DiscountTest.java

```diff
@@ -1,5 +1,43 @@
 package q4.team1;
 
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.util.ArrayList;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import q4.team1.exceptions.ItemNotFoundException;
+
 public class DiscountTest {
+    private ArrayList<CartItem> items;
+
+    private static class TestDiscount extends Discount {
```

Comment:
Good use of abstraction in your previous files, it is displayed well here.

Reply:
thank you kind sir